common/transport/transport-api/src/main/java/org/thingsboard/server/common/transport/service/DefaultTransportService.java

```diff
@@ -931,7 +931,7 @@ protected void processToTransportMsg(ToTransportMsg toSessionMsg) {
                     rateLimitService.remove(new DeviceId(entityUuid));
                     onDeviceDeleted(new DeviceId(entityUuid));
                 }
-            } else if (toSessionMsg.hasResourceUpdateMsg()) {
+            } else if (toSessionMsg.hasResourceUpdateMsg() && serviceInfoProvider.getServiceInfo().getTransportsList().contains(DataConstants.LWM2M_TRANSPORT_NAME)) {
```

Comment:
Why do we need this? I think this is redundant, since we won't receive such an update in the first place.